src/lightning/pytorch/strategies/fsdp.py

```diff
@@ -396,3 +401,42 @@ def register_strategies(cls, strategy_registry: _StrategyRegistry) -> None:
             cpu_offload=True,
         )
         cls._registered_strategies.append("fsdp_cpu_offload")
+
+    def load_optimizer_state_dict(self, checkpoint: Mapping[str, Any]) -> None:
+        optimizer_states = checkpoint["optimizer_states"]
+
+        assert self.model is not None
+        assert isinstance(self.model, FullyShardedDataParallel)
```

Comment:
The second assertion makes the first one redundant

```suggestion
        assert isinstance(self.model, FullyShardedDataParallel)
```